tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/AadAuthorityTests.cs

```diff
@@ -318,6 +318,31 @@ public void CanonicalAuthorityInitTest()
             Assert.AreEqual(UriCustomPortTailSlash, authority.AuthorityInfo.CanonicalAuthority);
         }
 
+        [TestMethod]
+        public void TenantSpecificAuthorityInitTest()
+        {
+            var host = String.Concat("https://", MsalTestConstants.ProductionPrefNetworkEnvironment);
+            var fullAuthority = String.Concat(host, "/" , MsalTestConstants.TenantId, "/");
+
+            var publicClient = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                             .WithAuthority(host, MsalTestConstants.TenantId)
+                                                             .BuildConcrete();
+
+            Assert.AreEqual(publicClient.Authority, fullAuthority);
+
+            publicClient = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                         .WithAuthority(host, new Guid(MsalTestConstants.TenantId))
```

Comment:
>.WithAuthority(host, new Guid(MsalTestConstants.TenantId)) [](start = 57, length = 58)

what is this testing that the above is not testing? Why new Guid?

Reply:
The above test uses a different constructor on the AuthorityInfo class that did not run into this bug which is why it did not fail. Also the withAuthority requires a Guid so I am simply passing a new GUID here that is created from our test guid string

Reply:
thanks

---
In reply to: [306517901](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1281#discussion_r306517901) [](ancestors = 306517901)